nwb_conversion_tools/testing/setup_paths.py

```diff
@@ -0,0 +1,38 @@
+from pathlib import Path
+import tempfile
+import os
+
+import pytest
+
+from nwb_conversion_tools.utils import load_dict_from_file
+
+
+# Load the configuration for the data tests
+test_config_dict = load_dict_from_file(Path(__file__).parent.parent.parent / "gin_test_config.json")
+
+#  GIN dataset: https://gin.g-node.org/CatalystNeuro/behavior_testing_data
+if os.getenv("CI"):
+    LOCAL_PATH = Path(".")  # Must be set to "." for CI
+    print("Running GIN tests on Github CI!")
+else:
+    # Override LOCAL_PATH in the `gin_test_config.json` file to a point on your system that contains the dataset folder
+    # Use DANDIHub at hub.dandiarchive.org for open, free use of data found in the /shared/catalystneuro/ directory
+    LOCAL_PATH = Path(test_config_dict["LOCAL_PATH"])
+    print("Running GIN tests locally!")
+BEHAVIOR_DATA_PATH = LOCAL_PATH / "behavior_testing_data"
+if not BEHAVIOR_DATA_PATH.exists():
+    pytest.fail(f"No folder found in location: {BEHAVIOR_DATA_PATH}!")
+
+ECEPHY_DATA_PATH = LOCAL_PATH / "ephy_testing_data"
+if not ECEPHY_DATA_PATH.exists():
+    pytest.fail(f"No folder found in location: {ECEPHY_DATA_PATH}!")
+
+OPHYS_DATA_PATH = LOCAL_PATH / "ophys_testing_data"
+if not OPHYS_DATA_PATH.exists():
+    pytest.fail(f"No folder found in location: {OPHYS_DATA_PATH}!")
+
+if test_config_dict["SAVE_OUTPUTS"]:
+    OUTPUT_PATH = LOCAL_PATH / "example_nwb_output"
+    OUTPUT_PATH.mkdir(exist_ok=True)
+else:
+    OUTPUT_PATH = Path(tempfile.mkdtemp())
```

Comment:
I like the idea of having all the variables defined in a file so the paths can be imported as a normal python variables. This is the first thing that I tried when I wrote the pull request with the configuration file. A clear advantage here is code reduction as this boilerplate is repeated more or less across the different gin tests file. I remember that @CodyCBakerPhD  was against this but sadly I can' t remember the reason. 

That aside, would it not make more sense semantically to have this somewhere in `/tests` though? Basically we intend for this to be the only place where it is used, correct?

Reply:
tried putting it in tests but wasn't able to get it to import. I'd be fine with putting it there if you can find a way to do it.

Reply:
@h-mayorquin some background on this.

A `testing` module is a module that is imported by tests and can also be imported by other packages. For example, `numpy.testing`, which contains functions like `assert_almost_equal`, which are useful for many different projects and therefore has been packaged within numpy instead of in the tests. Also, sklearn does something similar [here](https://github.com/scikit-learn/scikit-learn/blob/1495f6924/sklearn/utils/testing.py).

I think it's an open question here whether any other package might want to import these gin paths. If we did want to do that, we'd probably want to put the config file in a more accessible location that does not require `git clone`ing NCT, such as in `~/.nwb-conversion-tools/gin_config.json` but I don't want to do that work until there is another library that actually wants to use this.

Reply:
So you want to add __init__s to the testing module and put all of our source code inside a `src` file? Sounds good to me.

Reply:
Yes.

Maybe we can add the `__init__.py` to get the testing going here and do the `nwb_conversion_tools` to -> `src/nwb_conversion_tools` migration in a specific PR. What do you think?

Reply:
sounds good